breathecode/events/tests/urls/tests_academy_eventype_slug_visibilitysetting_id.py

```diff
@@ -0,0 +1,156 @@
+
+        response = self.client.get(url)
+        json = response.json()
+        expected = {'detail': 'Authentication credentials were not provided.', 'status_code': 401}
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 401)
+
+    def test_delete_visibilitysetting_with_bad_id(self):
+        self.bc.request.set_headers(academy=1)
+
+        url = reverse_lazy('events:academy_eventype_slug_visibilitysetting_id',
+                           kwargs={
+                               'event_type_slug': 'funny_event',
+                               'visibility_setting_id': 2
+                           })
+        self.generate_models(
+            authenticate=True,
+            profile_academy=1,
+            role=1,
+            capability='crud_event_type',
+            event_type={
+                'slug': 'funny_event',
+                'icon_url': 'https://www.google.com'
+            },
+            event_type_visibility_setting=True,
+        )
+
+        response = self.client.delete(url)
+        json = response.json()
+        expected = {'detail': 'event-type-visibility-setting-not-found', 'status_code': 404}
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 404)
+
+    def test_delete_visibilitysetting_with_bad_slug(self):
+        self.bc.request.set_headers(academy=1)
+
+        url = reverse_lazy('events:academy_eventype_slug_visibilitysetting_id',
+                           kwargs={
+                               'event_type_slug': 'funny_event',
+                               'visibility_setting_id': 1
+                           })
+        self.generate_models(
+            authenticate=True,
+            profile_academy=1,
+            role=1,
+            capability='crud_event_type',
+            event_type_visibility_setting=True,
+        )
+
+        response = self.client.delete(url)
+        json = response.json()
+        expected = {'detail': 'event-type-not-found', 'status_code': 400}
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 400)
+
+    def test_delete_visibilitysetting_with_no_other_event_type(self):
+        self.bc.request.set_headers(academy=1)
+
+        url = reverse_lazy('events:academy_eventype_slug_visibilitysetting_id',
+                           kwargs={
+                               'event_type_slug': 'funny_event',
+                               'visibility_setting_id': 1
+                           })
+        self.generate_models(
+            authenticate=True,
+            profile_academy=1,
+            role=1,
+            capability='crud_event_type',
+            event_type_visibility_setting=True,
+            event_type={
+                'slug': 'funny_event',
+                'icon_url': 'https://www.google.com'
+            },
+        )
+
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, 204)
```

Comment:
Add database test with
`self.bc.database.list_of('events.EventTypeVisibilitySetting')`

Reply:
Add a test to check the ids
```py
self.bc.check.queryset_with_pks(model.event_type.visibility_settings.all(), [2])
```

Where 2 is the primary key of resource added to many to many and `model.event_type.visibility_settings.all()` is the many to many to be checked